### ISSUE TITLE
Use credentials for tokens

### ISSUE THREAD

**Issue by hoshsadiq** [jenkinsci/generic-webhook-trigger-plugin#183]
The token option is useful and allows for targetting individual jobs. This token however also disables authentication on the Jenkins server for the triggers endpoint. It would be nice to have read from credentials rather than as a text input.

**Comment by tomasbjerre** [comment]
See also #165

**Comment by hoshsadiq** [comment]
Aah, pardon me. I did search but my searching skills clearly failed me.

Out of curiousity, are you still of the same opinion?

In my case I'm using job-dsl to generate the jobs and I set the triggers from the Job DSL rather than the jenkins pipeline. Because this plugin doesn't support credentials, I've not found any way of retrieving the token value in a safe manner, and I'm not really comfortable to check in the token into git.

**Comment by tomasbjerre** [comment]
It might be a good thing to implement.

I myself consider the token just something to identify the job, not something secret to add security. I would probably use the HMAC feature or implement some check within the pipeline script after triggering.

**Comment by hoshsadiq** [comment]
HMAC is definitely my go to, but there's certain scenarios where this is not possible and/or not easy. E.g. when implementing using webhooks for systems that aren't github/gitlab etc. In which case the token is the only thing that can be validated.

**Comment by tomasbjerre** [comment]
I will have another look at this when I find time.

**Comment by tomasbjerre** [comment]
I think you can avoid having the token in git by executing the Job DSL from a pipeline:

```groovy
withCredentials([string(credentialsId: 'mycredentialsid', variable: 'credentialsVariable')]) {
      jobDsl targets: 'work/jobs/*.groovy',
        removedJobAction: 'DELETE',
        removedViewAction: 'DELETE',
        lookupStrategy: 'SEED_JOB',
        additionalClasspath: "",
        additionalParameters: [token: credentialsVariable]
}
```

But still, would be nice if the token was not stored in the job, but instead only the credentials id was stored in the job.

**Comment by tomasbjerre** [comment]
Released in 1.68:

![image](https://user-images.githubusercontent.com/439571/94464846-4ef51d00-01bf-11eb-903e-3d8ca43919fb.png)

Open issue again if any problems.

**Comment by hoshsadiq** [comment]
Thank you very much @tomasbjerre!
